app/controllers/api/base_controller/action.rb

```diff
@@ -31,9 +31,16 @@ def api_resource(type, id, action_phrase)
         log_result(result)
         result
       rescue ActiveRecord::RecordNotFound, ForbiddenError, BadRequestError, NotFoundError => err
-        single_resource? ? raise : action_result(false, err.to_s)
+        raise if single_resource?
+        result = action_result(false, err.to_s)
+        add_href_to_result(result, type, id)
+        log_result(result)
+        result
```

Comment:
offhand this looks like it could be DRYed up with the code in the other rescue block.